language-design.md

```diff
@@ -28,6 +47,13 @@ you know a poorly designed language? What are the symptoms?
 
 **Response**
 
+Probably the most important and overarching symptom of a good programming language was discussed in class, and captured very well in a quote from the reading:
+> It should be easy to do simple things; possible to do complex things; and impossible, or at least difficult, to do wrong things. [Bloch 2006]
+
+All three of these are very important. If it is not easy to do simple things in a language, then users will never be willing to put in unnecessary work to do what would be easy in another language. The things that are categorized as simple may vary from language to language (or domain to domain), but there will always be simple operations that users understand should be easy. If complex things are impossible, then no one will ever be able to write more than a simple program; this may be acceptable for some little languages but is in general a sign that more features are necessary. Finally, most importantly for DSLs, the language needs to give the user a good sense of what things they should not do. If this is not the case then the language is likely poorly defined.
+
+These themes fit with other themes that we discussed as well. Good documentation is a good sign, while bad documentation may mean that the language was not completely conceived prior to being released. On a high level, if a language has little room for users to grow the language in a manner described by Steele, then it was likely not designed with users in mind, which can be a very major flaw. More low-level language design choices can also be clear symptoms of good or poor design: for instance, poor method names that oppose the user’s expectations are a very bad sign. 
+
 
```

Comment:
The documentation point is really important and definitely affects my choice of what languages/parts of languages to use. Do you have examples of what good documentation should include? Or examples of how documentation can be misleading (as opposed to just nonexistent)?